types/underscore/index.d.ts

```diff
@@ -109,17 +109,33 @@ declare module _ {
         null |
         undefined;
 
+    type ObjectIteratee<V, TResult> =
+        ObjectIterator<TypeOfDictionary<V, any>, TResult, V> |
+        EnumerableKey |
+        EnumerableKey[] |
+        object |
+        null |
+        undefined;
```

Comment:
It looks like they can, `Iteratee` (and `IterateeResult`) just get a little funkier. A diff of the funkiness does at least let us better see and discuss the differences, though 😄

Reply:
Due to [these comments](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/46422#discussion_r463329976), all the funkiness is gone, though the results are sadly slightly less cool since one ends up with `any` instead of `undefined` even if the type being evaluated definitely doesn't have the property.